src/Controls/src/Core/Shell/Shell.cs

```diff
@@ -1605,6 +1605,37 @@ void SendNavigated(ShellNavigatedEventArgs args)
 			CurrentItem?.Handler?.UpdateValue(Shell.TabBarIsVisibleProperty.PropertyName);
 		}
 
+		void PropagateSendNavigatedTo()
+		{
+			if (CurrentPage is null)
+			{
+				return;
+			}
+
+			// On Windows, the Loaded event has already fired (IsLoaded=true), so SendNavigatedTo runs immediately without altering the flow.
+			if (CurrentPage.IsLoaded)
+			{
+				CurrentPage.SendNavigatedTo(new NavigatedToEventArgs(_previousPage));
+			}
+			else
+			{
+				void OnCurrentPageLoaded(object sender, EventArgs e)
```

Comment:
If the page is removed/disposed before Loaded fires, the event handler will never be unsubscribed.